pkg/operator/controller/ttl_report.go

```diff
@@ -0,0 +1,87 @@
+package controller
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/aquasecurity/starboard/pkg/apis/aquasecurity/v1alpha1"
+	"github.com/aquasecurity/starboard/pkg/operator/etc"
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/api/errors"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type TTLReportReconciler struct {
+	logr.Logger
+	etc.Config
+	client.Client
+}
+
+func (r *TTLReportReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	err := ctrl.NewControllerManagedBy(mgr).
+		For(&v1alpha1.VulnerabilityReport{}).
+		Complete(r.reconcileReport())
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *TTLReportReconciler) reconcileReport() reconcile.Func {
+	return func(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+		log := r.Logger.WithValues("report", req.NamespacedName)
+
+		report := &v1alpha1.VulnerabilityReport{}
+		err := r.Client.Get(ctx, req.NamespacedName, report)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				log.V(1).Info("Ignoring cached report that must have been deleted")
+				return ctrl.Result{}, nil
+			}
+			return ctrl.Result{}, fmt.Errorf("getting report from cache: %w", err)
+		}
+		ttlReportAnnotationStr := report.Annotations[v1alpha1.TTLReportAnnotation]
+		if ttlReportAnnotationStr != "" {
+			creationTime := report.GetCreationTimestamp()
+			ttlExpired, durationToTTLExpiration, err := ttlIsExpired(ttlReportAnnotationStr, creationTime.Time)
+			if err != nil {
+				// TODO add a specific test to check if the error is that we can't convert to string
+				// Should be a custom error that we can use and point the user whats wrong. This should never happen... We controle the annotation
+				return ctrl.Result{}, err
+			}
+			if ttlExpired {
+				log.V(1).Info("Removing vulnerabilityReport")
+				r.Client.Delete(ctx, report, &client.DeleteOptions{})
+				// TODO Should we add a small delay I get "Ignoring cached report that must have been deleted", directly after a deletion
+				return ctrl.Result{Requeue: true}, nil
+			}
+			log.V(1).Info("RequeueAfter", "durationToTTLExpiration", durationToTTLExpiration)
+			return ctrl.Result{RequeueAfter: durationToTTLExpiration}, nil
+		}
+		return ctrl.Result{}, err
+	}
+}
+
+func ttlIsExpired(reportTTL string, creationTime time.Time) (bool, time.Duration, error) {
+	reportTTLInt, err := strconv.ParseInt(reportTTL, 10, 64)
```

Comment:
By using `time.ParseDuration` we can easily support all duration literal (20s, 15m, 1m20s), not just seconds.